[Hong Sheng] iP

loyhongshenggg opened the pull request:
# DukeTracker

Duke allows you to:
* track your tasks
* set deadlines
* set EVENT DATES 🤩


All you have to do is:
1. Log it into the GUI
2. Type simple commands (eg list, todo etc.)
3. and you're good to go!


If you're a java programmer, simply run the main function in Duke class!
```
public class Duke {
    public static void main(String[] args) {
        -- snip -- 
    }
}
```

What are you waiting for? Get to work!
- [x] Starts
- [x] Work
- [x] **Cute picture of doge** [here](https://i.kym-cdn.com/entries/icons/original/000/013/564/doge.jpg)
- [ ] NOW!

Commit:
Add Gradle support

loyhongshenggg committed:
Level-1

loyhongshenggg committed:
Level-2

loyhongshenggg committed:
Level-3

loyhongshenggg committed:
Level-4

loyhongshenggg committed:
A-TextUiTesting

loyhongshenggg committed:
Level-5

loyhongshenggg committed:
Level-6

loyhongshenggg committed:
Level-6

loyhongshenggg committed:
Level-7

loyhongshenggg committed:
branch-Level-8

loyhongshenggg committed:
Merge branch 'branch-Level-8'

loyhongshenggg committed:
A-MoreOOP

loyhongshenggg committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

loyhongshenggg committed:
A-Packages

loyhongshenggg committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

loyhongshenggg committed:
A-JUnit

loyhongshenggg committed:
A-JavaDoc

loyhongshenggg committed:
A-CodingStandard

loyhongshenggg committed:
A-CodingStandard

loyhongshenggg committed:
Level-9

sharmaine1028 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+        this.by = by;
+    }
+
+    protected void date_formatter() {
```
Can consider naming formatDate, since Java coding standards states that names representing methods must be verbs and written in camelCase

sharmaine1028 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    /**
+     * toString method that turns the input of deadline into a String type
```
Header comments for Javadoc should be present tense verb, e.g. Returns String representation of Deadline

sharmaine1028 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * toString method that turns the input of deadline into a String type
+     *
+     * @return String
```
Good to describe the return parameter, e.g. @return String representation of Deadline

sharmaine1028 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,64 @@
+        Parser parser = new Parser();
+
+        //process string
+        s = s.replace("[T]", "todo");
```
Why not abstract this to storage class?

sharmaine1028 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,65 @@
+     *
+     * @param object of type ArrayList<Task> which content is written to txt
+     */
+    public void writer(ArrayList<Task> arr) {
```
Consider naming to verb aka write

sharmaine1028 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,65 @@
+     *
+     * @return object of type String which is a String of saved information to be processed
+     */
+    public String reader() {
```
Consider naming to verb aka read

sharmaine1028 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+package duke;
+
+import duke.Task;
```
Take note of ordering of imports according to Java coding standards

sharmaine1028 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * to add items into TaskList
```
Capitalise and present tense verb e.g. Adds items into TaskList

sharmaine1028 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+
+public class TaskList {
+
+    int count = 0;
```
Remember to set access modifier explicitly for count

sharmaine1028 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Getter function to get the element in arr
```
Can consider changing header comment to Gets...

sharmaine1028 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,32 @@
+package duke;
+
+public class Ui {
+
```
Can remove these empty lines

sharmaine1028 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,32 @@
+    public String print(int i) {
+
+        switch (i) {
+            case 1: //bye
```
Take note of indentation for switch statements, no indentation for case clauses

sharmaine1028 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,32 @@
+                return "\nHere are the tasks in your list:";
+            case 3: //mark
+                return "Nice! I've marked this task as done:\n";
+            case 4:
```
What are these empty cases for?

sharmaine1028 commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,15 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Take note to order static imports first

sharmaine1028 reviewed (commented):
Looks alright, but some points to take note of:
1. Header comments for JavaDocs in present tense verb, "Adds..., Returns...,Gets..." and the first letter should be capitalised
2. Method names should be verbs and camelCase
3. Order of imports
4. Switch statement indentation

Otherwise, good job!

loyhongshenggg committed:
Level-10

loyhongshenggg committed:
Merge pull request #1 from loyhongshenggg/master

branch-Level-8

loyhongshenggg committed:
branch-A-Assertions

loyhongshenggg committed:
branch-A-Assertions

loyhongshenggg committed:
Revert "branch-A-Assertions"

This reverts commit f0d4a631794484975a962b6e2d3a282e7fa55348.

loyhongshenggg committed:
Merge pull request #2 from loyhongshenggg/branch-A-Assertions

branch-A-assertions

loyhongshenggg committed:
Placements of assert to various methods

There are several methods that contain assertions. These are the cases in which the user inputs something that is not expected.

By including assertions, the code will be less likely to have runtime errors.

loyhongshenggg committed:
Merge branch 'master' of https://github.com/loyhongshenggg/ip

loyhongshenggg committed:
Changes made to enhance code quality

Some refactoring of methods were made. Methods were refactored with the SLAP principle in mind to have added readability.

Comments were added to various classes to enhance readability.

To further improve readability, perhaps we could go one step further by using prominent layers of abstractions or even streams in subsequent code written.

loyhongshenggg committed:
Merge pull request #3 from loyhongshenggg/branch-A-CodeQuality

Changes made to enhance code quality

loyhongshenggg committed:
A-Assertions

loyhongshenggg committed:
A-BetterGui

loyhongshenggg committed:
Bug fix for parsing Deadline from storage

loyhongshenggg committed:
B-Snooze

loyhongshenggg committed:
minor changes for branch-Level-8

loyhongshenggg committed:
Final version, jar file has been created

loyhongshenggg committed:
README.md added

loyhongshenggg committed:
fixed README.md

loyhongshenggg committed:
README.md fix 2

loyhongshenggg committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/resources/META-INF/MANIFEST.MF
